boards/clue_nrf52840/src/main.rs

```diff
@@ -0,0 +1,491 @@
+//! Tock kernel for the Adafruit CLUE nRF52480 Express.
+//!
+//! It is based on nRF52840 Express SoC (Cortex M4 core with a BLE + IEEE 802.15.4 transceiver).
+
+#![no_std]
+// Disable this attribute when documenting, as a workaround for
+// https://github.com/rust-lang/rust/issues/62184.
+#![cfg_attr(not(doc), no_main)]
+#![feature(const_in_array_repeat_expressions)]
+#![deny(missing_docs)]
+
+use kernel::capabilities;
+use kernel::common::dynamic_deferred_call::{DynamicDeferredCall, DynamicDeferredCallClientState};
+use kernel::component::Component;
+use kernel::hil::gpio::Interrupt;
+use kernel::hil::i2c::I2CMaster;
+use kernel::hil::led::LedHigh;
+use kernel::hil::time::Counter;
+use kernel::hil::usb::Client;
+use kernel::mpu::MPU;
+use kernel::Chip;
+#[allow(unused_imports)]
+use kernel::{create_capability, debug, debug_gpio, debug_verbose, static_init};
+
+use nrf52840::gpio::Pin;
+use nrf52840::interrupt_service::Nrf52840DefaultPeripherals;
+
+// LEDs.
+const LED_RED_PIN: Pin = Pin::P1_01;
+const LED_WHITE_PIN: Pin = Pin::P0_10;
+
+const LED_KERNEL_PIN: Pin = Pin::P1_01;
+
+// Buttons
+const BUTTON_LEFT: Pin = Pin::P1_02;
+const BUTTON_RIGHT: Pin = Pin::P1_10;
+
+const GPIO_D2: Pin = Pin::P0_03;
+const GPIO_D3: Pin = Pin::P0_28;
+const GPIO_D4: Pin = Pin::P0_02;
+const GPIO_D6: Pin = Pin::P1_09;
+const GPIO_D7: Pin = Pin::P0_07;
+const GPIO_D8: Pin = Pin::P1_04;
+const GPIO_D9: Pin = Pin::P0_27;
+const GPIO_D10: Pin = Pin::P0_30;
+const GPIO_D12: Pin = Pin::P0_31;
+
+const _UART_TX_PIN: Pin = Pin::P0_05;
+const _UART_RX_PIN: Pin = Pin::P0_04;
+
+/// I2C pins for all of the sensors.
+const I2C_SDA_PIN: Pin = Pin::P0_24;
+const I2C_SCL_PIN: Pin = Pin::P0_25;
+
+/// Interrupt pin for the APDS9960 sensor.
+const APDS9960_PIN: Pin = Pin::P0_09;
+
+/// TFT ST7789H2
+const ST7789H2_SCK: Pin = Pin::P0_14;
+const ST7789H2_MOSI: Pin = Pin::P0_15;
+const ST7789H2_MISO: Pin = Pin::P0_26; // ST7789H2 has no MISO Pin, but SPI requires a MISO Pin
+const ST7789H2_CS: Pin = Pin::P0_12;
+const ST7789H2_DC: Pin = Pin::P0_13;
+const ST7789H2_RESET: Pin = Pin::P1_03;
+
+/// TFT backlight
+const _ST7789H2_LITE: Pin = Pin::P1_05;
+
+/// UART Writer for panic!()s.
+pub mod io;
+
+// State for loading and holding applications.
+// How should the kernel respond when a process faults.
+const FAULT_RESPONSE: kernel::procs::FaultResponse = kernel::procs::FaultResponse::Panic;
+
+// Number of concurrent processes this platform supports.
+const NUM_PROCS: usize = 8;
+
+static mut PROCESSES: [Option<&'static dyn kernel::procs::ProcessType>; NUM_PROCS] = [None; 8];
```

Comment:
You should use `NUM_PROCS` here as well instead of `8`.

Reply:
Fixed here and for nano ble 33